grommet/grommet
# fix: update "children" type

#### What does this PR do?
This PR updates the `children` type of the Checkbox to include `React.ReactNode`. This fixes a regression introduced when bringing in explicit typings for React Children. It also makes Checkbox consistent with the other changes in https://github.com/grommet/grommet/pull/6066 and with React's `PropsWithChildren` type signature ([link](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L800)).

#### Where should the reviewer start?
Compare the type update here to `RadioButton > children` and React's `PropsWithChildren`.

#### What testing has been done on this PR?
- tested locally in a linked environment
- manual testing steps (see below)

#### How should this be manually tested?
1. On the `master` branch, rename `Checkbox/stories/Children.js` to  `Checkbox/stories/Children.tsx`.
2. On lines 17-19 of `Checkbox/stories/Children.tsx`, substitute the function child component with a plain `React.ReactNode`. (e.g., `<span>foo</span>`). You should get a TypeScript error.
3. Stash that change, switch to this branch, and apply the change (or some equivalent -- we want the updated story file and the updated type definition on the same branch). You should no longer see a TypeScript error in `Checkbox/stories/Children.tsx`.

#### Do Jest tests follow these best practices?


#### Any background context you want to provide?
This issue was discovered because of a project that consumes Grommet and uses React's `PropsWithChildren` type.

#### What are the relevant issues?
N/A

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
N/A

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards-compatible
